internal/controller/serverset/serverset.go

```diff
@@ -218,25 +218,35 @@ func getLastCondition(server v1alpha1.Server) xpv1.Condition {
 	return xpv1.Condition{}
 }
 
-func fetchRole(ctx context.Context, e *external, replica v1alpha1.Server) v1alpha1.Role {
-	ns := "default"
-	if replica.Namespace != "" {
-		ns = replica.Namespace
+func fetchRole(ctx context.Context, e *external, sset v1alpha1.ServerSet, index int, replicaName, replicaStatus string) v1alpha1.Role {
+	role := v1alpha1.Passive
+	if replicaStatus != statusReady {
+		return role
 	}
-
+	if sset.Spec.ForProvider.IdentityConfigMap.Namespace == "" ||
+		sset.Spec.ForProvider.IdentityConfigMap.Name == "" ||
+		sset.Spec.ForProvider.IdentityConfigMap.KeyName == "" {
+		if index == 0 {
```

Comment:
If possible I think it might be helpful to log that it defaulted because there was no ConfigMap specified, in case of misconfiguration